Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -2438,11 +2445,14 @@ AODProducerWorkflowDPL::TrackExtraInfo AODProducerWorkflowDPL::processBarrelTrac
     extraInfoHolder.tpcNClsFindableMinusFound = tpcOrig.getNClusters() - tpcClData.found;
     extraInfoHolder.tpcNClsFindableMinusCrossedRows = tpcOrig.getNClusters() - tpcClData.crossed;
     extraInfoHolder.tpcNClsShared = tpcClData.shared;
-    if (src == GIndex::TPC) {                                                                                // standalone TPC track should set its time from their timebins range
-      double terr = 0.5 * (tpcOrig.getDeltaTFwd() + tpcOrig.getDeltaTBwd()) * mTPCBinNS;                     // half-span of the interval
-      double t = (tpcOrig.getTime0() + 0.5 * (tpcOrig.getDeltaTFwd() - tpcOrig.getDeltaTBwd())) * mTPCBinNS; // central value
-      LOG(debug) << "TPC tracks t0:" << tpcOrig.getTime0() << " tbwd: " << tpcOrig.getDeltaTBwd() << " tfwd: " << tpcOrig.getDeltaTFwd() << " t: " << t << " te: " << terr;
-      setTrackTime(t, terr, false);
+    if (src == GIndex::TPC) { // standalone TPC track should set its time from their timebins range
+      aod::track::extensions::TPCTimeErrEncoding p;
+      p.setDeltaTFwd(tpcOrig.getDeltaTFwd());
+      p.setDeltaTBwd(tpcOrig.getDeltaTBwd());
+      extraInfoHolder.trackTimeRes = p.getTimeErr();
+      extraInfoHolder.trackTime = tpcOrig.getTime0();
```

Comment:
@f3sch , the TPC time0 should be stored wrt the same reference as for other tracks, i.e. the BC of associated collision (if any) or the closest bc found by the fillBCSlice. And for that the t0 should be converted to ns.
So, I think the code should look like:
```
if (src == GIndex::TPC) { 
  double terr = 0.5 * (tpcOrig.getDeltaTFwd() + tpcOrig.getDeltaTBwd()) * mTPCBinNS;                     // half-span of the interval
  double t = (tpcOrig.getTime0() + 0.5 * (tpcOrig.getDeltaTFwd() - tpcOrig.getDeltaTBwd())) * mTPCBinNS; // central value
  if (needBCSlice) {
    double error = mTimeMarginTrackTime + terr;
    bcOfTimeRef = fillBCSlice(extraInfoHolder.bcSlice, t - error, t + error, bcsMap);
  }
  extraInfoHolder.trackTime = float(tpcOrig.getTime0()*mTPCBinNS - bcOfTimeRef * o2::constants::lhc::LHCBunchSpacingNS);
  aod::track::extensions::TPCTimeErrEncoding p;
  p.setDeltaTFwd(tpcOrig.getDeltaTFwd());
  p.setDeltaTBwd(tpcOrig.getDeltaTBwd());
  extraInfoHolder.trackTimeRes = p.getTimeErr();
  extraInfoHolder.diffBCRef = int(bcOfTimeRef);
  extraInfoHolder.isTPConly = true; // no truncation
  extraInfoHolder.flags |= o2::aod::track::TrackTimeAsym;
}
```
Also, for consistency, in the `getDeltaTFwd()` and `getDeltaTBwd()` I  would rescale the returned time to ns.

Reply:
Hi @shahor02, thanks this makes sense to me, when I used the timing in O2Phyiscs I wrote custom code to get correct BC, but this is nicer. For the deltas are they not already in ns when I return them encoding, or am I missing something?

Reply:
No, both the time0 and the deltas are in TPC time-bins.

Reply:
Ah right, but for this I would need, right?
```
    // RS FIXME: this is not yet fetched from the CCDB
    auto& elParam = o2::tpc::ParameterElectronics::Instance();
    mTPCBinNS = elParam.ZbinWidth * 1.e3;
```

I do not think that I can do something like this in a getter for the column. Otherwise I would have to do that already before doing the encoding.

Reply:
You can safely use `constexpr float TPCBinNS = 8*o2::constants::lhc::LHCBunchSpacingNS;`, I don't think we will level change the TPC TB definition.

Reply:
Ok, thanks!